src/Ocelot.Provider.Consul/Consul.cs

```diff
@@ -14,19 +14,22 @@ public class Consul : IServiceDiscoveryProvider
 
     public Consul(ConsulRegistryConfiguration config, IOcelotLoggerFactory factory, IConsulClientFactory clientFactory)
     {
-        _logger = factory.CreateLogger<Consul>();
         _config = config;
         _consul = clientFactory.Get(_config);
+        _logger = factory.CreateLogger<Consul>();
     }
 
-    public async Task<List<Service>> Get()
+    public const string ServiceValidationWarningFormat = "Unable to use service address: '{0}' and port: {1} as it is invalid for the service: '{2}'. Address must contain host only e.g. 'localhost', and port must be greater than 0.";
```

Comment:
Do we really want to expose this string, it's imho, error prone?  Maybe it would be wise to have a factory as a static method?
```c#
public static string GetServiceValidationWarning(string address, int port, string serviceName)
    {
        return $"Unable to use service address: '{address}' and port: {port} as it is invalid for the service: '{serviceName}'. Address must contain host only e.g. 'localhost', and port must be greater than 0.";
    }
```

Reply:
What is the goal of such wrapping if this method will have one reference in the class where it is used?
Definition of new method makes sense if there are 2+ references.
The message is reused by one user scenario. This scenario is covered by a unit test where we check actual value.

Second, if we define a public method, we have to unit test it!
I'm not a fan to write one additional separated unit test.

Reply:
Yeah, but it is the same problem if you are using a public string and reuse it later in unit tests... eg.
```string.Format(ConsulProvider.ServiceValidationWarningFormat, service.Address, service.Port, service.Service);```

Reply:
But at the end, it doesn't really matter. I just don't like the fact of having a string with parameters as public variable.

Reply:
✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/eefa04272e6b8f435408f8eef23132194b7706d7